pyaerocom/helpers.py

```diff
@@ -140,6 +132,13 @@ def extract_latlon_dataarray(arr, lat, lon, lat_dimname=None,
         lat = [lat]
     if isinstance(lon, str):
         lon = [lon]
+    if check_domain:
+        arr_lat = arr[lat_dimname].data
+        arr_lon = arr[lon_dimname].data
+        lat0, lat1 = arr_lat.min(), arr_lat.max()
+        lon0, lon1 = arr_lon.min(), arr_lon.max()
+        lat = [x for x in lat if lat0 <= x <= lat1]
+        lon = [x for x in lon if lon0 <= x <= lon1]
```

Comment:
I think this should be changed so that only pairs of latitudes and longitudes that exists gets added to lat and lon.
Currently this might lead to lat and lon having different lengths which causes problems in line 146.
Something like this (not tested):
`lat, lon = [x,y for x,y in zip(lat, lon) if (lat0 <= x <= lat1) and (lon0 <= y <= lon1) ]`

Reply:
@ejgal That is correct! Thanks for looking into this. Can you fix and send a new PR into v090dev?

Reply:
Yup, will fix.

Reply:
@ejgal I moved this into a new issue #127 since this PR is merged.